src/engine/source/builder/src/builders/stage/indexerOutput.cpp

```diff
@@ -9,6 +9,26 @@
 namespace builder::builders
 {
 
+namespace
+{
+
+// lowercase field conversion and spaces replacement with hyphens
+auto sanitizeField = [](const std::string& field, const base::Event& event) -> std::string
+{
+    auto fieldValue = event->getString(field);
+    if (fieldValue == std::nullopt)
+    {
+        throw std::runtime_error(fmt::format("Field '{}' does not exist in the event", field));
+    }
+    auto fieldString = fieldValue.value();
+    std::transform(
+        fieldString.begin(), fieldString.end(), fieldString.begin(), [](unsigned char c) { return std::tolower(c); });
+    std::replace(fieldString.begin(), fieldString.end(), ' ', '-');
+    return fieldString;
+};
```

Comment:
- `fieldValue` is a `opt`, move the string to avoid copies on return.
- Iterate through the string only once.
 - Let's not use tolower in this case, it's slow because it queries the locale.  [ref](https://en.cppreference.com/w/cpp/string/byte/tolower.html)
- Avoid allocating memory, concatenating, and creating a string that is not used as a trace.

```suggestion
auto sanitizeField = [](std::string_view field, const base::Event& event) -> std::string
{
    auto opt = event->getString(field);
    if (!opt){
            throw std::runtime_error("Missing field");
    }


    std::string s = std::move(*opt);
    for (unsigned char& c : s)
    {
        if (c >= 'A' && c <= 'Z')
            c += 'a' - 'A';
        else if (c == ' ')
            c = '-';
    }
    return s;
};
```

Reply:
Great, applied suggestion